inst/@sym/eye.m

```diff
@@ -0,0 +1,86 @@
+%% Copyright (C) 2016 Lagu
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod @@sym eye (@var{n})
+%% @defmethodx @@sym eye (@var{n}, @var{m})
+%% @defmethodx @@sym eye (@var{n}, @var{m}, @var{class})
+%% Return an identity matrix.
+%%
+%% Example:
+%% @example
+%% @group
+%% y = eye (sym(3))
+%%   @result{} y = (sym 3×3 matrix)
+%%  ⎡1  0  0⎤
+%%  ⎢       ⎥
+%%  ⎢0  1  0⎥
+%%  ⎢       ⎥
+%%  ⎣0  0  1⎦
+%% @end group
+%% @end example
+%%
+%% @seealso{@@sym/zeros, @@sym/ones}
+%% @end defmethod
+
+%% Source: http://docs.sympy.org/dev/modules/matrices/matrices.html
+
+function y = eye(varargin)
+
+
+  % partial workaround for issue #13: delete when/if fixed properly
+  if (strcmp (varargin{nargin}, 'sym'))
+    nargin = nargin - 1;
+    varargin = varargin(1:nargin);
+  end
+
+  if (isa (varargin{nargin}, 'char'))
+    y = eye (cell2nosyms (varargin){:});
+    return
+  end
+
+  y = python_cmd ('return eye(*_ins)', varargin{:});
+
+end
+
+%!test
+%! y = eye(sym(2));
+%! x = [1 0;0 1];
+%! assert( isequal( y, sym(x)))
+
+%!test
+%! y = eye(sym(2), 1);
+%! x = [1;0];
```

Comment:
Sorry, I meant it constructively.  I very much appreciate all your edits, hope you don't feel otherwise.  Just trying to help make them even better (and lower my review time ;-)